.gitignore

```diff
@@ -1,27 +1,106 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
+
+# I don't konw.
 _trial_temp
-build
-build-pypy
-MANIFEST
-dist
 apidocs
 *,cover
 .testrepository
-*.pyc
-*.pyd
-*.pyo
-*.so
 *~
-*.swp
-*.swh
-*.swn
-*.swo
+*.sw[phno]
 docs/tutorial/index.html
 dulwich.egg-info/
+docs/api/*.txt
+dulwich.dist-info
+
+# Unit test / coverage reports
+htmlcov/
 .tox/
+.coverage
+.coverage.*
+.cache
+nosetests.xml
+coverage.xml
+*.cover
+.hypothesis/
+.pytest_cache/
 .idea/
 .coverage
-htmlcov/
-docs/api/*.txt
+
+# Distribution / packaging
+.Python
+build/
+develop-eggs/
+dist/
+downloads/
+eggs/
+.eggs/
+lib/
+lib64/
+parts/
+sdist/
+var/
+wheels/
+*.egg-info/
+.installed.cfg
+*.egg
+MANIFEST
+
+# PyInstaller
+#  Usually these files are written by a python script from a template
+#  before PyInstaller builds the exe, so as to inject date/other infos into it.
+*.manifest
+*.spec
+
+# Installer logs
+pip-log.txt
+pip-delete-this-directory.txt
+
+# Scrapy stuff:
+.scrapy
+
+# Sphinx documentation
+docs/_build/
+
+# PyBuilder
+target/
+
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# pyenv
+.python-version
+
+# celery beat schedule file
+celerybeat-schedule
+
+# SageMath parsed files
+*.sage.py
+
+# Environments
```

Comment:
What creates / uses these files?

Reply:
Please refer to the following

[venv — Creation of virtual environments](https://docs.python.org/3/library/venv.html)

And I strongly recommend to keep this, this is very useful for most developers, I also use this to avoid interference between kits